### DISCUSSION TITLE
Swagger Enpoint Visibility

### DISCUSSION THREAD

**Discussion by hpposch** [elsa-workflows/elsa-core#2971] [Q&A]
In the current preview all workflows that start with a HttpEndpoint Activity are displayed as swagger Enpoints.
This is a really nice feature and helps me a lot.
I was just wondering if you plan to split the 'Workflow Endpoints' from all the other Endpoints that are visible in swagger.
Or maybe a seperate swagger json that holds only the workflow endpoints.
From my point of view it is kind of a security issue when they show up together.
I was also wondering if it is possible to 'lock' or 'hide' the internal Endpoints?

Best regards and thanks for this great feature ;)

**Comment by sfmskywalker** [comment]
I hadn't planned for it, but I like the suggestion to be able to split the workflow **HTTP Endpoint** activity endpoints from the rest. We could probably make this aspect configurable. Do you want to create an issue for it? 😃 

With regards to hiding the other API endpoints implemented as API controllers, I'm not so sure what to do here, other than annotating them with `[Authorize]` attributes and a specific policy name, allowing you to configure these policies from `Startup` (or `Program` as it is the case these days with .NET 6) so that they are at least not exposed anonymously.

**Comment by hpposch** [comment]
Sure, i will create an issue for that.

I think it would be enough to configure the visibility in code.
I mean it would be a killer feature, to define the visibility directly on the endpoint task, but i guess this would be kind of difficult.

**Reply by sfmskywalker** [reply]
Actually, this would be very easy. And given that there are other aspects we may want to be able to control with swagger doc generation, I think we may want to add a "Swagger" or "Open API" tab to the HTTP Endpoint activity to allow for further control. We could add an "Enabled" checkbox here that the user can toggle to control whether this endpoint is included or not.

**Reply by hpposch** [reply]
That would be great!
